### ISSUE TITLE
why is the BLEU score of the translate result  so bad？

### ISSUE THREAD

**Issue by qtxue** [jadore801120/attention-is-all-you-need-pytorch#93]
I can gain 50.8% accuracy on training set and 40.6% accuracy on validation set with WMT14 ch-en,but the translate result can only gain 1% BLEU score,and i find many sentence have the same beginning words or Phrases.Have you test the BLEU score when you got the model?
[ Epoch 9 ]
  - (Training)   ppl:  42.43964, accuracy: 50.856 %, elapse: 98.557 min                                                                                                                                                                          
  - (Validation) ppl:  33.06820, accuracy: 40.699 %, elapse: 0.026 min

**Comment by GZJAS** [comment]
I also found the similar questions...but I didn't know why

**Comment by dksehdals216** [comment]
What is WMT'14 ch-en? I am not aware of such a data set.

**Comment by qtxue** [comment]
> What is WMT'14 ch-en? I am not aware of such a data set.

Sorry,I was wrong,my data set is got from NIST.

**Comment by qlwang25** [comment]
@qtxue @GZJAS 
I also get the similar problem.
the size of dataset may  cause this phenomenon.

> We trained on the standard WMT 2014 English-German dataset consisting of about 4.5 million
sentence pairs

u can find this sentence in **Attention Is All You Need**.
And, u can download [dataset](https://nlp.stanford.edu/projects/nmt/)

**Comment by zhao1402072392** [comment]
could i have the code of caculating bleu score part,i have some problem at that,thanks^_^

**Comment by jadore801120** [comment]
Sorry for the confusion, I made the beam search now more concise now.
Please take a look on the newest code, Thanks.
